### ISSUE TITLE
Attempting to compile with MSVC, failure on MCCS

### ISSUE THREAD

**Issue by strega-nil** [ocaml/opam#3186]
I've followed the opam compilation instructions for MSVC, using cygwin and `make cold CONFIGURE_ARGS="--prefix /cygdrive/c/ocaml"`, and I get the error:

```
bfd.c(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\glpk.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\env.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\stdc.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\bfd.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\fvs.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\fhvint.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\fhv.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\luf.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\sva.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\lufint.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\sgf.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\scfint.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\scf.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\btf.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\ifu.h(1): error C2019: expected preprocessor directive, found '1'
c:\users\ubsan\source\langs\opam\_build\default\src_ext\mccs\src\btfint.h(1): error C2019: expected preprocessor directive, found '1'
```

The reason for this error is:

```c
# 1 "src_ext/mccs/src/glpk/bflib/btf.h"
```

on the first line of all of the glpk files.

**Comment by ubsan** [comment]
cc @dra27

**Comment by dra27** [comment]
You need to use the master version of jbuilder (and the master version of ocaml-mccs)

**Comment by dra27** [comment]
Incidentally, the mingw and msvc builds of opam in master are uninteresting at the moment - they (should) build, but they don't (quite yet) do anything!

**Comment by ubsan** [comment]
@dra27 ah, okay! well, thanks, that's kinda unfortunate :P

I'm very excited for when I can finally just use ocaml on windows :P

**Comment by dra27** [comment]
If you want to use OCaml, there are various ways - one of which is to execute `make OCAML_PORT=msvc64 compiler` in opam, which will build a full OCaml in the bootstrap directory.

**Comment by ubsan** [comment]
@dra27 currently I'm using wsl; I mostly just want to switch off of wsl as soon as possible, but I do require opam before doing that.

**Comment by dra27** [comment]
@ubsan - Windows opam won't eliminate the need for either WSL or Cygwin, though you won't have to use either of them directly (you should then be able to use opam either from the Command Prompt or PowerShell)

**Comment by ubsan** [comment]
@dra27 hmm? what would require cygwin, beyond the makefile?

**Comment by kit-ty-kate** [comment]
I believe this was fixed in opam 2.2.0 which brought native windows support. If this is not the case please do reopen an issue.
